### PR TITLE
inotify-info: init at unstable-2024-01-05

### PR DESCRIPTION
**Pull request by motiejus**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/in/inotify-info/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "mikesart";
+    repo = pname;
+    rev = "a7ff6fa62ed96ec5d2195ef00756cd8ffbf23ae1";
+    sha256 = "sha256-yY+hjdb5J6dpFkIMMUWvZlwoGT/jqOuQIcFp3Dv+qB8=";
+  };
+
+  installPhase = ''
+    mkdir -p "$out/bin"
+    cp "_release/inotify-info" "$out/bin/"
```
Comment (jtbx) [inline pkgs/by-name/in/inotify-info/package.nix:16]: `runHook` should be used in phases like this one. If a user wants to override this package to add a `preInstall` or `postInstall` phase, then that phase will be called if it exists.
```suggestion
    runHook preInstall
    install -Dm755 _release/inotify-info $out/bin/inotify-info
    runHook postInstall
```

File: pkgs/by-name/in/inotify-info/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "inotify-info";
+  version = "unstable-2024-01-06";
+
+  src = fetchFromGitHub {
+    owner = "mikesart";
+    repo = pname;
+    rev = "a7ff6fa62ed96ec5d2195ef00756cd8ffbf23ae1";
+    sha256 = "sha256-yY+hjdb5J6dpFkIMMUWvZlwoGT/jqOuQIcFp3Dv+qB8=";
```
Comment (jtbx) [inline pkgs/by-name/in/inotify-info/package.nix:11]:
```suggestion
    hash = "sha256-yY+hjdb5J6dpFkIMMUWvZlwoGT/jqOuQIcFp3Dv+qB8=";
```

File: pkgs/by-name/in/inotify-info/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    mkdir -p "$out/bin"
+    cp "_release/inotify-info" "$out/bin/"
+  '';
+
+  meta = with lib; {
+    description = "Easily track down the number of inotify watches, instances, and which files are being watched.";
+    homepage = "https://github.com/mikesart/inotify-info";
+    license = licenses.mit;
+    platforms = platforms.linux;
+    maintainers = [ maintainers.motiejus ];
```
Comment (jtbx) [inline pkgs/by-name/in/inotify-info/package.nix:24]:
```suggestion
    maintainers = with maintainers; [ motiejus ];
```

**Review by jtbx — CHANGES_REQUESTED** [review]

**Comment by motiejus** [comment]
Thank you so much @jtbx, this is excellent feedback.

File: nixos/doc/manual/release-notes/rl-2405.section.md [hunk 4]
```diff
@@ -257,6 +257,8 @@ The pre-existing [services.ankisyncd](#opt-services.ankisyncd.enable) has been m
 
 - `libass` now uses the native CoreText backend on Darwin, which may fix subtitle rendering issues with `mpv`, `ffmpeg`, etc.
 
+- [inotify-info](https://github.com/mikesart/inotify-info), a command-line utility to track inotify watches, was added.
+
```
Comment (jtbx) [inline nixos/doc/manual/release-notes/rl-2405.section.md:261]: I don't think new packages are meant to go in a changelog

**Comment by motiejus** [comment]
Sorry, just messed up a rebase. One moment.

**Comment by motiejus** [comment]
> Sorry, just messed up a rebase. One moment.

Done, ready for review.

**Review by jtbx — APPROVED** [review]
Looks good to me :+1:

**Comment by jtbx** [comment]
Oh, one thing, please squash b337caa and 4a69080 into one commit named "inotify-info: init at unstable-2024-01-05"

**Review by jtbx — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -12819,6 +12819,16 @@
     githubId = 10601196;
     name = "Jérémie Ferry";
   };
```
Comment (drupol) [inline maintainers/maintainer-list.nix:12821]: Sort attributes alphabetically please

File: pkgs/by-name/in/inotify-info/package.nix [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    hash = "sha256-yY+hjdb5J6dpFkIMMUWvZlwoGT/jqOuQIcFp3Dv+qB8=";
+  };
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 _release/inotify-info $out/bin/inotify-info
+    runHook postInstall
+  '';
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/by-name/in/inotify-info/package.nix:20]: Sort attributes alphabetically and add meta.mainProgram

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

I've left some comments, let me know if this is clear enough.

**Review by drupol — APPROVED** [review]

**Comment by motiejus** [comment]
Thank you!

**Pull request merged**